File: backend/car-catalog-backend/src/main/java/com/wswork/carcatalog/repository/MarcaRepository.java
package com.wswork.carcatalog.repository;

import com.wswork.carcatalog.entity.Marca;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface MarcaRepository extends JpaRepository<Marca, Long> {
    
    // Buscar marca por nome
    Optional<Marca> findByNomeMarca(String nomeMarca);
    
    // Verificar se existe marca com o nome
    boolean existsByNomeMarca(String nomeMarca);
}

File: backend-kotlin/car-catalog-kotlin/bin/main/com/wswork/carcatalog/repository/MarcaRepository.kt
package com.wswork.carcatalog.kotlin.repository

import com.wswork.carcatalog.entity.Marca
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface MarcaRepository : JpaRepository<Marca, Long> {
    fun findByNomeMarca(nomeMarca: String): Marca?
    fun existsByNomeMarca(nomeMarca: String): Boolean
}
